pkg/volume/csi/csi_test.go

```diff
@@ -304,19 +304,30 @@ func TestCSI_VolumeAll(t *testing.T) {
 				}
 
 				// creates VolumeAttachment and blocks until it is marked attached (done by external attacher)
-				go func(spec *volume.Spec, nodeName types.NodeName) {
-					attachID, err := volAttacher.Attach(spec, nodeName)
-					if err != nil {
-						t.Errorf("csiTest.VolumeAll attacher.Attach failed: %s", err)
-						return
-					}
-					t.Logf("csiTest.VolumeAll got attachID %s", attachID)
-
-				}(volSpec, host.GetNodeName())
+				var attachID string
+				var attachErr error
+				attachDone := make(chan struct{})
+				go func() {
+					defer close(attachDone)
+					attachID, attachErr = volAttacher.Attach(volSpec, host.GetNodeName())
+				}()
 
 				// Simulates external-attacher and marks VolumeAttachment.Status.Attached = true
 				markVolumeAttached(t, host.GetKubeClient(), fakeWatcher, attachName, storage.VolumeAttachmentStatus{Attached: true})
 
+				// Wait for async attach op to finish. Check for errors.
+				select {
+				case <-attachDone:
+					if attachErr != nil {
+						t.Errorf("csiTest.VolumeAll attacher.Attach failed: %s", attachErr)
+						return
+					}
+					t.Logf("csiTest.VolumeAll got attachID %s", attachID)
+				case <-time.After(500 * time.Millisecond):
```

Comment:
It might not be safe to assume the `Attach()` call could finish in 500ms. I think we could use `sync.WaitGroup` here.